Rails 4.2 update

mkasztelnik opened the pull request.

mkasztelnik committed:
Up to rails 4.2.1

mkasztelnik committed:
Dependencies update

mkasztelnik committed:
Rename outdated configuration key

`config.serve_static_assets` renamed into `config.serve_static_files`

mkasztelnik committed:
Guard configuration

mkasztelnik committed:
Fix tests after rails 4.2 update

mkasztelnik commented:
@VPH-Share/cyfronet-dev-team please take a look, it is deployed into dev environment

nowakowski commented on app/serializers/basic_endpoint_serializer.rb:
```diff
@@ -2,9 +2,12 @@
   def url
-    atmosphere.descriptor_api_v1_endpoint_url(object.id)
+    atmosphere.
+      descriptor_api_v1_endpoint_url(object.id)
```
Why the line split?

mkasztelnik replied:
Right, I will revert it. It was done during active model serializer 0.8.3 tests.

mkasztelnik replied:
Fixed

nowakowski commented:
:+1:

mkasztelnik committed:
Use rails 4.2.2.beta because of bug with not serving HEAD response

Take a look here
  * https://github.com/rails/rails/issues/18698
  * https://github.com/rails/rails/pull/19431
for problem description. You can track second link to track when the
problem is fixed and merged.

mkasztelnik commented:
I needed to switch into rails `4-2-stable` branch, since in `4.2.1` there is a [bug](https://github.com/rails/rails/issues/18698) connected with not responding to `HEAD` request. Currently commit which introduces this bug is rolled back from this branch and [appropriate fix](https://github.com/rails/rails/pull/19431) is waiting to be merged. Most probably this will be fixed in rails `4.2.2`, but since it will be released we can use stable `4.2` branch.

mkasztelnik committed:
Fix problem with I18n on founds view
